core/client/src/main/java/alluxio/client/block/stream/PacketOutStream.java

```diff
@@ -253,24 +253,26 @@ private void updateCurrentPacket(boolean lastPacket) throws IOException {
       return;
     }
 
-    if (mCurrentPacket.readableBytes() > 0) {
-      if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
-        try {
+    if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
```

Comment:
Then I will need to duplicate the logic of sending the packets since I need to send the last packet as well if it contains any data. That doesn't look clean to me. Let me know if I didn't understand your suggestion correctly.